crates/uv-resolver/src/resolver/urls.rs

```diff
@@ -25,155 +33,132 @@ impl Urls {
         git: &GitResolver,
         dependencies: DependencyMode,
     ) -> Result<Self, ResolveError> {
-        let mut urls: FxHashMap<PackageName, VerbatimParsedUrl> = FxHashMap::default();
+        let mut urls: FxHashMap<PackageName, Vec<VerbatimParsedUrl>> = FxHashMap::default();
+        let mut overrides: FxHashMap<PackageName, VerbatimParsedUrl> = FxHashMap::default();
 
         // Add all direct requirements and constraints. If there are any conflicts, return an error.
-        for requirement in manifest.requirements(markers, dependencies) {
-            match &requirement.source {
-                RequirementSource::Registry { .. } => {}
-                RequirementSource::Url {
-                    subdirectory,
-                    location,
-                    url,
-                } => {
-                    let url = VerbatimParsedUrl {
-                        parsed_url: ParsedUrl::Archive(ParsedArchiveUrl::from_source(
-                            location.clone(),
-                            subdirectory.clone(),
-                        )),
-                        verbatim: url.clone(),
-                    };
-                    if let Some(previous) = urls.insert(requirement.name.clone(), url.clone()) {
-                        if !Self::same_resource(&previous.parsed_url, &url.parsed_url, git) {
-                            return Err(ResolveError::ConflictingUrlsDirect(
-                                requirement.name.clone(),
-                                previous.verbatim.verbatim().to_string(),
-                                url.verbatim.verbatim().to_string(),
-                            ));
+        for requirement in manifest.requirements_no_overrides(markers, dependencies) {
```

Comment:
Why no overrides?

Reply:
They have different semantics and are handled below

Reply:
Why?

Reply:
I added better inline docs about that at `Urls`